docs/internal/types.md

```diff
@@ -67,7 +67,64 @@ Most importantly, capabilities are not type traits like C++'s `const int * const
 
 ## Module / Class model
 
-TODO
+In Verona, modules are classes.
+Classes and functions inside modules are just nested types and methods of the module's class.
+
+Modules are composed of all Verona files in a specific directory.
+But there is no special relationship between modules in hierarchical directory structure.
+Sub-directories are not part of the module, nor visible in any special way to the module in the parent directory.
+
+For example, the following directory structure:
+```
+myproject +-- main.verona
+          |-- driver.verona
+          \-- Module1 +-- foo.verona
+          |           '-- bar.verona
+          \-- Module2 +-- baz.verona
+                      |-- ber.verona
+                      \-- Module3 +-- mod3.verona
+```
+
+translates to the following module structure:
+```ts
+class myproject {
+  // both main and driver inserted here
+}
+class Module1 {
+  // both foo and bar inserted here
+}
+class Module2 {
+  // both baz and ber inserted here
+}
+class Module3 {
+  // mod3 inserted here
+}
+```
```

Comment:
After today's deeply rabbit-holed meeting, I think it might be worth expanding the "inserted here"s with some example text and, perhaps, giving examples of paths that do and don't reach, from within each module, the other modules.  Examples like "Inside `Module2/baz.verona`, the name `"Module3"` finds `Module3` (under the `./` search path) and the name `"Module1"` finds `Module1` (under the `$project_root` search path) which is also found by the name `"../Module1"` (under the `./` search path)."

Reply:
I've updated the document with what seems to have been the consensus on the meeting today. Package discovery and relative paths (and its perils).